hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplaySyncReplicationWALCallable.java

```diff
@@ -108,23 +113,21 @@ private void replayWAL(String wal) throws IOException {
           request.getReplicationClusterId(), request.getSourceBaseNamespaceDirPath(),
           request.getSourceHFileArchiveDirPath());
         // Read next entries.
-        entries = readWALEntries(reader);
+        entries = readWALEntries(reader, wal);
       }
+    } finally {
+      reader.close();
     }
   }
 
-  private Reader getReader(String wal) throws IOException {
+  private WALStreamReader getReader(String wal) throws IOException {
```

Comment:
Why didn't we use TailingReader in this replication scenario?

Reply:
This is WAL replay. Although it is part of the sync replication implementation, it is not for 'replicating' data, it does not need to tail the WAL file, just read the file once and replay the edits.